catalog/spatial/registry/registry-publication-action-provider/src/main/java/org/codice/ddf/registry/publication/action/provider/RegistryPublicationActionProvider.java

```diff
@@ -0,0 +1,262 @@
+                    .map(Object::toString)
+                    .collect(Collectors.toList());
+        }
+        if (event.getTopic()
+                .equals(CREATED_TOPIC) || event.getTopic()
+                .equals(UPDATED_TOPIC)) {
+            publications.put(registryId, locations);
+        } else if (event.getTopic()
+                .equals(DELETED_TOPIC)) {
+            publications.remove(registryId);
+        }
+    }
+
+    public void init() throws FederationAdminException {
+        List<Metacard> metacards = federationAdminService.getRegistryMetacards();
+        for (Metacard mcard : metacards) {
+            Attribute locations =
+                    mcard.getAttribute(RegistryObjectMetacardType.PUBLISHED_LOCATIONS);
+            if (locations != null) {
+                publications.put(mcard.getAttribute(RegistryObjectMetacardType.REGISTRY_ID)
+                                .getValue()
+                                .toString(),
+                        locations.getValues()
+                                .stream()
+                                .map(Object::toString)
+                                .collect(Collectors.toList()));
+            } else {
+                publications.put(mcard.getAttribute(RegistryObjectMetacardType.REGISTRY_ID)
+                        .getValue()
+                        .toString(), Collections.emptyList());
+            }
+        }
+    }
+
+    private Action getAction(String regId, String destinationId, boolean publish) {
+
+        URL url;
+        String title = publish ? PUBLISH_TITLE + destinationId : UNPUBLISH_TITLE + destinationId;
+        String description = publish ?
+                PUBLISH_DESCRIPTION + destinationId :
+                UNPUBLISH_DESCRIPTION + destinationId;
+        String operation = publish ? PUBLISH_OPERATION : UNPUBLISH_OPERATION;
+        String httpOp = publish ? HTTP_POST : HTTP_DELETE;
+        try {
+
+            String path = String.format("%s/%s/%s/%s",
+                    REGISTRY_PATH,
+                    regId,
+                    PUBLICATION_PATH,
+                    URLEncoder.encode(destinationId, "UTF-8"));
+            URI uri = new URI(SystemBaseUrl.constructUrl(path, true));
+            url = uri.toURL();
+
+        } catch (MalformedURLException | URISyntaxException | UnsupportedEncodingException e) {
+            LOGGER.error("Malformed URL exception", e);
+            return null;
+        }
+        String id = String.format("%s.%s.%s", getId(), operation, httpOp);
+        return new ActionImpl(id, title, description, url);
+    }
+
+    private <T> String getRegistryId(T subject) {
+        if (subject instanceof Metacard) {
+            Metacard mcard = (Metacard) subject;
+            if (!mcard.getTags()
+                    .contains(RegistryConstants.REGISTRY_TAG)) {
+                return null;
+            }
+            return ((Metacard) subject).getAttribute(RegistryObjectMetacardType.REGISTRY_ID)
```

Comment:
you should be able to use your `mcard` variable here instead of doing another case on `subject`